### PR TITLE
A rework of RISC-V build survey

### PR DESCRIPTION
**Pull request by Icenowy**
Topic Description
-----------------

Reworking #3439 .

Package(s) Affected
-------------------

```
aspell
mariadb
libftdi
ibus
qt-5
xfsprogs
graphviz
qt-4
```

Security Update?
----------------

No

Build Order
-----------

Any like the original survey. 

Architectural Progress
----------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

File: extra-libs/libftdi/autobuild/beyond [hunk 1]
```diff
@@ -1,3 +1,4 @@
 abinfo "Moving misplaced files in /usr/lib64 ..."
+mkdir -pv "$PKGDIR"/usr/lib/
 cp -rv "$PKGDIR"/usr/lib64/* "$PKGDIR"/usr/lib/
```
Comment (MingcongBai) [inline extra-libs/libftdi/autobuild/beyond]: You might want to add an if branch here.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-graphics/graphviz/autobuild/beyond [hunk 2]
```diff
@@ -1,3 +1,5 @@
-abinfo "Moving misplaced files under /lib to /usr/lib ..."
-cp -rv "$PKGDIR"/lib/* "$PKGDIR"/usr/lib/
-rm -rv "$PKGDIR"/lib
+if [ -d "$PKGDIR"/lib ]; then
```
Comment (MingcongBai) [inline extra-graphics/graphviz/autobuild/beyond:1]: Or lib64? Does it happen?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-libs/libftdi/autobuild/beyond (hunk 1 above)
Comment (Icenowy) [reply extra-libs/libftdi/autobuild/beyond]: what if branch? check whether /usr/lib64 exists?

File: extra-graphics/graphviz/autobuild/beyond (hunk 2 above)
Comment (Icenowy) [reply extra-graphics/graphviz/autobuild/beyond:1]: Well, I think it's files from Tcl, so I think it does not happen, at least on riscv64/amd64.

File: extra-libs/libftdi/autobuild/beyond (hunk 1 above)
Comment (MingcongBai) [reply extra-libs/libftdi/autobuild/beyond]: Yes

File: extra-graphics/graphviz/autobuild/beyond (hunk 2 above)
Comment (MingcongBai) [reply extra-graphics/graphviz/autobuild/beyond:1]: Cool.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**